amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -111,6 +114,7 @@ public TaskResult call() {
                     // Create new lease for the child shards if they don't exist.
                     if (!CollectionUtils.isNullOrEmpty(childShards)) {
                         createLeasesForChildShardsIfNotExist();
+                        updateLeasesForChildShards();
                     } else {
                         log.warn("Shard {} no longer exists. Shutting down consumer with SHARD_END reason without creating leases for child shards.", shardInfoIdProvider.apply(shardInfo));
```

Comment:
Can you add more comments here on why we permit this flow to go ahead, even when mandatory childshards are missing? Also, can we revisit the ResourceNotFound scenario and check if we can avoid that from happening?

Reply:
The ResourceNotFound exception is catched in KinesisDataFetcher and FanoutRecordsPublisher. For this exception, we don't throw exception, Instead we send out the shardend signal to trigger the shutdown task with shutdown reason as SHARD_END, and populate the GetRecordsResponse with empty list of childShards. I think we should avoid changing the current behavior which is to shutdown the consumers when the stream is missing. This can happen if the customer deletes the stream while leaving the KCL application running.